kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/ViewBackHandler.kt

```diff
@@ -0,0 +1,65 @@
+package com.squareup.workflow.ui
+
+import android.content.Context
+import android.content.ContextWrapper
+import android.view.View
+import android.view.View.OnAttachStateChangeListener
+import androidx.activity.ComponentActivity
+import androidx.activity.OnBackPressedCallback
+
+typealias BackPressHandler = () -> Unit
+
+var View.backPressedHandler: BackPressHandler?
+  get() {
+    val handler = getTag(R.id.view_back_handler)
+    check(handler is Function0<*>?) {
```

Comment:
Why is this check necessary here? Isn't it enough to just return null like all the other getters? Even this getter uses Elvis right below.

Reply:
It's an invariant check, e.g. to make sure some other code hasn't stomped our field. Anyway, point could be moot if I can move it out of the view tag like you suggested elsewhere.

Reply:
None of the other getters do invariant checks though, they just do the safe cast, which is why I'm wondering why this one is special.

Reply:
I was stupidly overreacting to the loss of the un-reified return type. I'm going to follow up and move this all into a single view tag like you suggested, that should moot the issue anyway.